### PR TITLE
Open Lineage Services: Major exception-handling refactoring

### PR DESCRIPTION
**Pull request by GuyAckermans**

### CODE REVIEW

**Comment by GuyAckermans** [comment]
@planetf1 DCO sign-off seems to fail for correctly signed commits, any suggestions?

**Comment by planetf1** [comment]
> @planetf1 DCO sign-off seems to fail for correctly signed commits, any suggestions? The recommended steps of git commit --amend --signoff don't work (never works for me)

When I looked in the list of commits in the PR, The following don't appear to be signed off:
* f15b8fe3c576f0d887c2c04ab02019408a35241a
* 805aecfe57633ce000a1e6621b1b5b7925be140c

All commits need to be signed to pass the DCO, so you will need to rebuild the PR and re-signoff the commits

**Comment by planetf1** [comment]
the merge commit may be ok actually

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-client/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/client/OpenLineageClient.java [hunk 1]
```diff
@@ -66,9 +68,10 @@ public LineageVerticesAndEdges lineage(String userId,
 
     private void detectExceptions(String methodName,
                                   LineageResponse response)
-            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException {
-        exceptionHandler.detectAndThrowInvalidParameterException(methodName, response);
-        exceptionHandler.detectAndThrowPropertyServerException(methodName, response);
+            throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException, OpenLineageException {
```
Comment (ruxandraRosu) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-client/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/client/OpenLineageClient.java]: Please note that when backend server is up, but not configured as OLS, response is null and a NPE is thrown after detectExceptions method

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/util/OpenLineageExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Throw an detectAndThrowOpenLineagexception if it is encoded in the REST response.
+     *
+     * @param methodName name of the method called
+     * @param restResult response from the rest call.  This generated in the remote server.
+     * @throws PropertyServerException encoded exception from the server
+     */
+    public void detectAndThrowOpenLineagexception(String methodName,
```
Comment (lcpopa) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-api/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/util/OpenLineageExceptionHandler.java]: Not a biggie, just a typo

**Pull request merged**